### PR TITLE
book features

### PR DESCRIPTION
**Pull request by HiIAmTzeKean**
user can now search book using controller

### CODE REVIEW

File: src/main/java/seedu/dukeofbooks/common/IVerifiable.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package seedu.dukeofbooks.common;
+
+public interface IVerifiable {
+    public abstract boolean equals(Object other);
+    public abstract int hashCode();
+    public abstract String toString();
```
Comment (danxuZhang) [inline src/main/java/seedu/dukeofbooks/common/IVerifiable.java:6]: A minor issue: it is redundant to write `public` and `abstract` for methods in an interface. From [Oracle Java Documentation](https://docs.oracle.com/javase/specs/jls/se7/html/jls-9.html#jls-9.4):  

> It is permitted, but discouraged as a matter of style, to redundantly specify the public and/or abstract modifier for a method declared in an interface.

You can directly write:
``` java
public interface IBlahBlah {
    int foo();
    void bar();
}

```

File: src/main/java/seedu/dukeofbooks/data/book/Book.java [hunk 2]
```diff
@@ -32,5 +33,25 @@ public Person getAuthor() {
     public void setAuthor(Person author) {
         this.author = author;
     }
-
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof Book // instanceof handles nulls
+                && this.hasSameData((Book) other));
```
Comment (danxuZhang) [inline src/main/java/seedu/dukeofbooks/data/book/Book.java:40]: This boolean expression is a bit too long? 
Suggests: 
``` java
if (other == this) {
  return true;
} else {  // if is not the same object
  other instanceof Book && hasSameData
}

Comment (danxuZhang) [reply src/main/java/seedu/dukeofbooks/data/book/Book.java:40]: This is also for other `equals`.

**Review by danxuZhang — COMMENTED** [review]
Looks good! Just some minor issues need to modify. 👍

File: src/main/java/seedu/dukeofbooks/data/book/Book.java (hunk 2 above)
Comment (HiIAmTzeKean) [reply src/main/java/seedu/dukeofbooks/data/book/Book.java:40]: Was thinking that the expression was clear, but will make the changes for greater code readability.

**Comment by HiIAmTzeKean** [comment]
fixed comments on a997cb5

**Pull request merged**